test/extensions/stats_sinks/hystrix/hystrix_test.cc

```diff
@@ -0,0 +1,348 @@
+    setCountersToZero();
+  }
+
+  // Attach the counter to cluster_stat_scope and set default value.
+  void setCounterForTest(NiceMock<Stats::MockCounter>& counter, std::string counter_name) {
+    counter.name_ = counter_name;
+    ON_CALL(cluster_stats_scope_, counter(counter_name)).WillByDefault(ReturnRef(counter));
+  }
+
+  void setCountersToZero() {
+    ON_CALL(error_5xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(retry_5xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(error_4xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(retry_4xx_counter_, value()).WillByDefault(Return(0));
+    ON_CALL(success_counter_, value()).WillByDefault(Return(0));
+  }
+
+  NiceMock<Upstream::MockCluster> cluster_;
+  Upstream::MockClusterInfo* cluster_info_ = new NiceMock<Upstream::MockClusterInfo>();
+  Upstream::ClusterInfoConstSharedPtr cluster_info_ptr_{cluster_info_};
+
+  NiceMock<Stats::MockStore> stats_store_;
+  NiceMock<Stats::MockStore> cluster_stats_scope_;
+  const std::string cluster_name_;
+
+  NiceMock<Stats::MockGauge> membership_total_gauge_;
+  NiceMock<Stats::MockCounter> success_counter_;
+  NiceMock<Stats::MockCounter> error_5xx_counter_;
+  NiceMock<Stats::MockCounter> retry_5xx_counter_;
+  NiceMock<Stats::MockCounter> error_4xx_counter_;
+  NiceMock<Stats::MockCounter> retry_4xx_counter_;
+};
+
+class HystrixSinkTest : public testing::Test {
+public:
+  HystrixSinkTest() { sink_.reset(new HystrixSink(server_, window_size_)); }
+
+  // Return the value corresponding to key in an event stream.
+  absl::string_view getStreamField(absl::string_view data_message, absl::string_view key) {
+    absl::string_view::size_type key_pos = data_message.find(key);
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view trim_data_before_Key = data_message.substr(key_pos);
+    key_pos = trim_data_before_Key.find(" ");
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view trim_data_after_value = trim_data_before_Key.substr(key_pos + 1);
+    key_pos = trim_data_after_value.find(",");
+    EXPECT_NE(absl::string_view::npos, key_pos);
+    absl::string_view actual = trim_data_after_value.substr(0, key_pos);
+    return actual;
+  }
+
+  // Return a string without quotes.
+  absl::string_view getStringStreamField(absl::string_view data_message, absl::string_view key) {
+    absl::string_view value = getStreamField(data_message, key);
+    return value.substr(1, value.length() - 2);
+  }
+
+  Buffer::OwnedImpl createClusterAndCallbacks() {
+    // Set cluster.
+    cluster_map_.emplace(cluster1_name_, cluster1_.cluster_);
+    ON_CALL(server_, clusterManager()).WillByDefault(ReturnRef(cluster_manager_));
+    ON_CALL(cluster_manager_, clusters()).WillByDefault(Return(cluster_map_));
+
+    Buffer::OwnedImpl buffer;
+    auto encode_callback = [&buffer](Buffer::Instance& data, bool) {
+      // Set callbacks to send data to buffer.
+      buffer.add(
```

Comment:
this still looks awkward. I was thinking of:
```
     // Set callbacks to send data to buffer. This will append to the end of the
     // buffer, so multiple calls will all be dumped one after another into this buffer.
     buffer.add(data);